### ISSUE TITLE
Does not clean `ghc-*.tar.bz2` files

### ISSUE THREAD

**Issue by andreasabel** [juhp/stack-clean-old#9]
Looks like the downloaded GHC archives are not cleaned up together with their installation.  E.g. I see:
```
$ l /Users/abel/.stack/programs/x86_64-osx/
total 7363096
drwxr-xr-x  34 abel  staff       1088 Aug 17 10:21 .
drwxr-xr-x   3 abel  staff         96 Jan  2  2019 ..
drwxr-xr-x   5 abel  staff        160 Jan 16  2021 ghc-8.0.2
-rw-------   1 abel  staff          9 Jan 16  2021 ghc-8.0.2.installed
-rw-r--r--   1 abel  staff  186822039 Jan 16  2021 ghc-8.0.2.tar.bz2
-rw-r--r--   1 abel  staff  277789010 Jan 12  2021 ghc-8.10.3.tar.bz2
-rw-r--r--   1 abel  staff  277873042 Feb 22  2021 ghc-8.10.4.tar.bz2
-rw-r--r--   1 abel  staff  274146402 May 19 11:22 ghc-8.10.5.tar.bz2
drwxr-xr-x   5 abel  staff        160 Sep 19  2021 ghc-8.10.7
-rw-------   1 abel  staff          9 Sep 19  2021 ghc-8.10.7.installed
-rw-r--r--   1 abel  staff  274034358 Sep 19  2021 ghc-8.10.7.tar.bz2
drwxr-xr-x   5 abel  staff        160 Oct 26  2020 ghc-8.2.2
-rw-------   1 abel  staff          9 Oct 26  2020 ghc-8.2.2.installed
-rw-r--r--   1 abel  staff  192467920 Oct 26  2020 ghc-8.2.2.tar.bz2
drwxr-xr-x   5 abel  staff        160 Mar 15  2019 ghc-8.4.4
-rw-r--r--   1 abel  staff          9 Mar 15  2019 ghc-8.4.4.installed
-rw-r--r--   1 abel  staff  230734167 Mar 15  2019 ghc-8.4.4.tar.bz2
drwxr-xr-x   5 abel  staff        160 Oct 26  2020 ghc-8.6.5
-rw-------   1 abel  staff          9 Oct 26  2020 ghc-8.6.5.installed
-rw-r--r--   1 abel  staff  226259652 Oct 26  2020 ghc-8.6.5.tar.bz2
-rw-------   1 abel  staff          9 Jul  5  2021 ghc-8.8.2.installed
-rw-r--r--   1 abel  staff  272458737 Jul  5  2021 ghc-8.8.2.tar.bz2
drwxr-xr-x   5 abel  staff        160 Dec 25  2020 ghc-8.8.4
-rw-------   1 abel  staff          9 Dec 25  2020 ghc-8.8.4.installed
-rw-r--r--   1 abel  staff  272402324 Dec 25  2020 ghc-8.8.4.tar.bz2
-rw-r--r--   1 abel  staff  275378217 Feb 22  2021 ghc-9.0.1.tar.bz2
drwxr-xr-x   6 abel  staff        192 Jan 31  2022 ghc-9.0.2
-rw-------   1 abel  staff          9 Jan 31  2022 ghc-9.0.2.installed
-rw-r--r--   1 abel  staff  264948596 Jan 31  2022 ghc-9.0.2.tar.bz2
-rw-r--r--   1 abel  staff  172033065 Mar 10 11:06 ghc-9.2.1.tar.bz2
-rw-r--r--   1 abel  staff  279562928 Apr  2 20:56 ghc-9.2.2.tar.bz2
drwxr-xr-x   6 abel  staff        192 Aug 17 10:11 ghc-9.2.4
-rw-------   1 abel  staff          9 Aug 17 10:11 ghc-9.2.4.installed
-rw-r--r--   1 abel  staff  278448084 Aug 17 10:09 ghc-9.2.4.tar.bz2
```

**Comment by juhp** [comment]
Do you mean the older minor version tarballs should be removed?  I think I agree with that.

**Comment by andreasabel** [comment]
Would be great to have this feature!

**Comment by juhp** [comment]
Okay, that should be possible: originally it was intentional since tarballs take up less space and can be time-consuming or difficult to re-download in some situations. Perhaps it should be an option or separate command even possibly.

**Comment by juhp** [comment]
I tried to implement it, based off the ghc installations handling code.

Can you give it try?  I only really tested listing.

**Comment by andreasabel** [comment]
If there is only the tarball left, it does not find it...  E.g. the tarball was left over on my system after cleaning 9.4.5.
```
drwxr-xr-x   6 abel  staff        192 Jun 10 12:27 ghc-9.2.8
-rw-------   1 abel  staff          9 Jun 10 12:28 ghc-9.2.8.installed
-rw-r--r--   1 abel  staff  280241113 Jun 10 12:26 ghc-9.2.8.tar.bz2
-rw-r--r--   1 abel  staff  296264786 Jun 10 12:10 ghc-9.4.5.tar.bz2
drwxr-xr-x   5 abel  staff        160 Aug 21 14:31 ghc-9.4.6
-rw-------   1 abel  staff          9 Aug 21 14:33 ghc-9.4.6.installed
-rw-r--r--   1 abel  staff  296491411 Aug 21 14:27 ghc-9.4.6.tar.bz2
drwxr-xr-x   5 abel  staff        160 Jun 22 21:24 ghc-9.6.2
-rw-------   1 abel  staff          9 Jun 22 21:26 ghc-9.6.2.installed
-rw-r--r--   1 abel  staff  309039548 Jun 22 21:21 ghc-9.6.2.tar.bz2

$ stack-clean-old remove 9.4.5 -C
stack ghc compiler version 9.4.5 not found
```

**Comment by juhp** [comment]
Hmm I can't reproduce that - I did some testing now with dummy files.
I think the tarball code should only be looking at the tarballs (`ghc-*.tar.*`).

**Comment by juhp** [comment]
Ah wait: there are separate options now for --compilers and --tarballs.
Maybe it is only me who wants to keep tarballs around for posterity?
Though I was pondering to add a combined options too (--programs??):
maybe I should allow multiple options instead hmm.

I am not really in love with all the options/naming...
Something like --installations and --tarballs (combining into --compilers?) could be better though also confusing.
Naming is hard and all that. I may sleep on it until 0.5.

**Comment by juhp** [comment]
This should be in the released [0.4.7](https://hackage.haskell.org/package/stack-clean-old-0.4.7)
